global/scripts/mppmodecheck.shlib

```diff
@@ -0,0 +1,231 @@
+    else
+      log_Msg "-> Independent processing of BOLD image (HCPStyleData)"
+    fi
+
+    # -- Use of distortion correction methods
+
+    if [ "${DistortionCorrection}" == 'NONE' ]; then
+      log_Msg "-> Set not to use any distortion correction method (LegacyStyleData)"
+      Compliance="LegacyStyleData"
+    else
+      log_Msg "-> Distortion correction method specified (${DistortionCorrection}; HCPStyleData)"
+    fi
+
+    # -- Use of BOLD mask
+
+    if [ ! "${Mask}" == 'T1' ]; then
+      log_Msg "-> Set to use mask: ${Mask} (LegacyStyleData)"
+      Compliance="LegacyStyleData"
+    else
+      log_Msg "-> Set to use T1w image based mask (HCPStyleData)"
+    fi
+
+    # -- Preregister code
+
+    if [ ! "${Preregister}" == 'epi_reg' ]; then
+      Compliance="LegacyStyleData"
+      log_Msg "-> Set to use ${Preregister} for bold image pre-registration (LegacyStyleData)"
+    else
+      log_Msg "-> Set to use ${Preregister} for bold image pre-registration (HCPStyleData)"
+    fi
+
+    # -- BOLD registration to reference image
+
+    if [ ! "${ReferenceReg}" == 'linear' ]; then
+      Compliance="LegacyStyleData"
+      log_Msg "-> Set to use ${ReferenceReg} registration to reference image (LegacyStyleData)"
+    else
+      log_Msg "-> Set to use ${ReferenceReg} registration to reference image (HCPStyleData)"
+    fi 
+}
+
+check_mpp_fMRISurface()
+{
+    log_Msg "-> No LegacyStyleData functionality"
+}
+
+check_mpp_Diffusion()
+{
+    log_Msg "-> No LegacyStyleData functionality"   
+}
+
+
+
+#  --- Main function for checking compliance
+# 
+#  Use: check_mpp_compliance <pipeline name>
+
+
+check_mpp_compliance()
+{
+
+    # --- Define variables
+
+    local pipeline=$1
+    local hcpversion=`cat "$HCPPIPEDIR"/version.txt`
+    local Compliance="HCPStyleData"
+    local ispipeline="false"
+
+    # --- check and abort if pipeline name is not known
+
+    check_pipeline_name
+
+    if [ "${ispipeline}" == "false" ]; then
+        log_Err_Abort "${pipeline} is not a valid HCP Pipeline name. Aborting execution!"
+    fi
+
+    # --- Initial information
+
+    if [ "${MPPMode}" = "LegacyStyleData" ]; then
+      log_Msg "LegacyStyleData Minimal Preprocessing Pipelines: ${pipeline} ${version}"
+      log_Msg "NOTICE: You are using MPP mode that enables processing of images that do not"
+      log_Msg "        conform to the HCP specification as described in Glasser et al. (2013)!"
+      log_Msg "        Be aware that if the HCP requirements are not met, the level of data quality"
+      log_Msg "        can not be guaranteed and the Glasser et al. (2013) paper should not be used"
+      log_Msg "        in support of this workflow. A manuscript with comprehensive evaluation for"
+      log_Msg "        the Extended MPP workflow is in active preparation and should be appropriately"
+      log_Msg "        cited when published."
+      log_Msg "        "
+      log_Msg "Checking available data and processing options"
+    else
```

Comment:
It isn't clear to me that we should print anything if the mode and compliance are both hcp.

Reply:
The check starts with:
> Checking data compliance with HCPStyleData HCP MPP mode

Wouldn't it make sense to report that the check was successful?

Reply:
I would also remove the message stating that it is starting checking anything, and really all messages except either reporting failure, or activation of legacy mode (and also report only the non-compliant parameters, or that legacy mode is active with full hcp compliance).

My personal style is that messages can be long if necessary, but if a message doesn't say something that deserves someone's attention, then it shouldn't exist.

Reply:
Done.